### PR TITLE
Update registration tests

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds some test coverage for update registration.

### Type of change

- [ ] bug fix
- [ ] new feature
- [ ] documentation
- [x] testing

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/bff/models/v1/registration_test.go [hunk 1]
```diff
@@ -576,6 +576,125 @@ func TestValidateTRISA(t *testing.T) {
...
+	update := &RegistrationForm{}
+	err := loadJSONFixture("testdata/registration_form.json", update)
+	require.NoError(t, err, "error loading registration form fixture")
+
+	// An error should be returned if an unknown step is provided
+	err = form.Update(update, "invalid")
+	require.EqualError(t, err, "unknown step \"invalid\"", "error should be returned for unknown step")
+
+	// Test updating the basic details step
+	update.State.Current = 1
```
Comment (bbengfort) [inline pkg/bff/models/v1/registration_test.go:602]: Is there an error if the state is not on the correct step? 

Also - I think we should be returning the step name in the state, but we haven't been doing that; is there a story to remediate that?

**Review by bbengfort — APPROVED** [review]

File: pkg/bff/models/v1/registration_test.go (hunk 1 above)
Comment (pdeziel) [reply pkg/bff/models/v1/registration_test.go:602]: We currently are not checking the state on the backend side, should we be?

Comment (bbengfort) [reply pkg/bff/models/v1/registration_test.go:602]: No we shouldn't be - I was just wondering why the tests were setting current to the correct step. But if it's what the front-end is doing then it's probably good to have in our tests. 

As for the step names; the backend is currently the only thing that knows those names -- but we can leave state entirely to the front-end if that makes more sense.

Comment (pdeziel) [reply pkg/bff/models/v1/registration_test.go:602]: I was just setting it to something to ensure that the state is being updated on the form, but we have been relying on the frontend to manage when the state is changed.

Comment (bbengfort) [reply pkg/bff/models/v1/registration_test.go:602]: Ok, sounds good

**Pull request merged**